src/OpenApi/test/OpenApiGeneratorTests.cs

```diff
@@ -88,17 +88,17 @@ static void AssertCustomRequestFormat(OpenApiOperation operation)
         }
 
         AssertCustomRequestFormat(GetOpenApiOperation(
-            [Consumes("application/custom")] (InferredJsonClass fromBody) => { }));
+            [Consumes("application/custom")] ([FromQuery] int foo) => { }));
 
         AssertCustomRequestFormat(GetOpenApiOperation(
-            [Consumes("application/custom")] ([FromBody] int fromBody) => { }));
+            [Consumes("application/custom")] ([FromHeader] int fromBody) => { }));
     }
 
     [Fact]
     public void AddsMultipleRequestFormatsFromMetadata()
     {
         var operation = GetOpenApiOperation(
-            [Consumes("application/custom0", "application/custom1")] (InferredJsonClass fromBody) => { });
+            [Consumes("application/custom0", "application/custom1")] (int foo) => { });
 
         var request = Assert.Single(operation.Parameters);
```

Comment:
```suggestion
        var request = Assert.Empty(operation.Parameters);
```

Let's keep the route handler the same here too.